### PR TITLE
Add __repr__ and __str__ to Metrics baseclass

### PR DESCRIPTION
**Pull request by MathijsdeBoer**
### Description

When training a model using MONAI metrics for experiment tracking, I tend to log which metrics I am using. Unfortunately, just sending the metrics objects to Tensorboard will result in a list like [CustomMetric1, CustomMetric2, <monai.metrics.Metric ...>, etc.]

Adding `__repr__` and `__str__` methods to the base class will solve this small annoyance. The current implementation will only return the class name, but if a certain metric would wish to report more data for its `__repr__` string, this can be easily overridden in any subclass.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by MathijsdeBoer** [comment]
Ideally one would exist for the Loss classes, too. But the baseclass for them is `torch.nn.modules.loss._Loss`, so either one could make a subclass of that class, add the `__repr__` and `__str__` methods to it, and point all existing loss classes to the new one instead, or these methods could be added to the baseclass in PyTorch with a pull request in that project.

File: monai/metrics/metric.py [hunk 1]
```diff
@@ -37,6 +37,12 @@ def __call__(self, *args: Any, **kwargs: Any) -> Any:
         """
         raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
 
+    def __repr__(self):
+        return self.__class__.__name__
```
Comment (KumoLiu) [inline monai/metrics/metric.py:41]: Thanks for the PR!
It would be beneficial to have something akin to <monai.metrics.meandice.DiceMetric object at 0x7f7b820c08e0>, as it will yield more detailed information pertaining to DiceMetric. Despite this, I would still recommend utilizing the original implementation, or alternatively, consider the addition of the __str__ method for simplification.

What do you think? cc @ericspod @Nic-Ma

Comment (MathijsdeBoer) [reply monai/metrics/metric.py:41]: What additional important information does the text `<monai.metrics.meandice.DiceMetric object at 0x7f7b820c08e0>` give over `DiceMetric`?

Comment (ericspod) [reply monai/metrics/metric.py:41]: The purpose of `__repr__` is to provide an unambiguous representation of an object, rather than `__str__` which is a more readable and descriptive representation. With the original `__repr__` we get the object ID which allows us to tell multiple instances of an object apart, this is important during debugging or testing. I would say that returning the class name only is ambiguous since it lacks any way of certainly telling instances apart.

File: monai/metrics/metric.py [hunk 2]
```diff
@@ -37,6 +37,12 @@ def __call__(self, *args: Any, **kwargs: Any) -> Any:
...
+
+    def __str__(self):
```
Comment (ericspod) [inline monai/metrics/metric.py:43]: Here `__str__` is producing something readable and concise which is fine, in other contexts it might make sense to include some state information if this wasn't too lengthy.

Comment (MathijsdeBoer) [reply monai/metrics/metric.py:40]: I've removed `__repr__`, just having the `__str__` will reduce a bit of code for me either way!

File: monai/metrics/metric.py (hunk 1 above)
Comment (MathijsdeBoer) [reply monai/metrics/metric.py]: `__repr__` removed, makes sense that one would like to have more detailed information from that call.

**Review by KumoLiu — APPROVED** [review]
